services/user/Fractals/query-service/src/lib.rs

```diff
@@ -0,0 +1,292 @@
+            FractalTable::with_service(fractals::SERVICE)
+                .get_index_pk()
+                .get(&fractal)
+        }
+
+        async fn evaluations(&self, fractal: AccountNumber) -> Vec<EvaluationInstance> {
+            EvaluationInstanceTable::with_service(fractals::SERVICE)
+                .get_index_pk()
+                .range((fractal, 0)..=(fractal, u32::MAX))
+                .collect()
+        }
+
+        async fn scores(
+            &self,
+            fractal: AccountNumber,
+            member: Option<AccountNumber>,
+        ) -> Vec<Score> {
+            ScoreTable::with_service(fractals::SERVICE)
+                .get_index_pk()
+                .range(
+                    (fractal, 0, member.unwrap_or(AccountNumber::new(0)))
+                        ..=(
+                            fractal,
+                            u32::MAX,
+                            member.unwrap_or(AccountNumber::new(u64::MAX)),
+                        ),
+                )
+                .collect()
+        }
+
+        async fn fractals(
+            &self,
+            first: Option<i32>,
+            last: Option<i32>,
+            before: Option<String>,
+            after: Option<String>,
+        ) -> async_graphql::Result<Connection<RawKey, Fractal>> {
+            TableQuery::new(FractalTable::with_service(fractals::SERVICE).get_index_pk())
+                .first(first)
+                .last(last)
+                .before(before)
+                .after(after)
+                .query()
+                .await
+        }
+
+        async fn fractals_list(&self, fractals: Vec<AccountNumber>) -> Vec<Option<Fractal>> {
+            fractals
+                .into_iter()
+                .map(|account| {
+                    FractalTable::with_service(fractals::SERVICE)
+                        .get_index_pk()
+                        .get(&account)
+                })
+                .collect()
+        }
+
+        async fn member(&self, fractal: AccountNumber, member: AccountNumber) -> Option<Member> {
+            MemberTable::with_service(fractals::SERVICE)
+                .get_index_pk()
+                .get(&(fractal, member))
+        }
+
+        async fn memberships(
+            &self,
+            member: AccountNumber,
+            first: Option<i32>,
+            last: Option<i32>,
+            before: Option<String>,
+            after: Option<String>,
+        ) -> async_graphql::Result<Connection<RawKey, Member>> {
+            TableQuery::subindex::<AccountNumber>(
+                MemberTable::with_service(fractals::SERVICE).get_index_by_member(),
+                &(member),
+            )
+            .first(first)
+            .last(last)
+            .before(before)
+            .after(after)
+            .query()
+            .await
+        }
+
+        async fn fractal_memberships(&self, member: AccountNumber) -> Vec<Fractal> {
```

Comment:
This needed?

Reply:
It's being used on the UI to render the fractal switcher

Reply:
This has now been deleted, fractal switcher now makes use of the new GraphQL feature you introduced.